astarte/device/interface.py

```diff
@@ -74,27 +74,44 @@ def __init__(self, interface_definition: dict):
         ValueError
             if both version_major and version_minor numbers are set to 0
         """
-        self.name: str = interface_definition["interface_name"]
-        self.version_major: int = interface_definition["version_major"]
-        self.version_minor: int = interface_definition["version_minor"]
+        self.name: str = interface_definition.get("interface_name")
+        self.version_major: int = interface_definition.get("version_major")
+        self.version_minor: int = interface_definition.get("version_minor")
+        self.type: str = interface_definition.get("type")
+        self.ownership = interface_definition.get("ownership")
+
+        if not (
+            isinstance(self.name, str)
+            and isinstance(self.version_major, int)
+            and isinstance(self.version_minor, int)
+            and self.type in {"datastream", "properties"}
+            and self.ownership in (DEVICE, SERVER)
+        ):
+            raise InterfaceFileDecodeError("Error parsing one interface.")
```

Comment:
Maybe here we could add some info about the interface to make it easier to know which interface is the root of the problem ? The name can be printed (even if it could be an object different from a string) or we could dump part of the json to keep it simple. Even printing all the known properties that are checked in the if statement could be an idea.

Reply:
The interface name could be `None` and I don't think that would be useful.
I think that dumping the `json` can help.